sdk/servicebus/azure-servicebus/samples/async_samples/receive_deadlettered_messages_async.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+Example to show receiving dead-lettered messages from a Service Bus Queue asynchronously.
+"""
+
+# pylint: disable=C0111
+
+import os
+import asyncio
+from azure.servicebus.aio import ServiceBusClient
+
+CONNECTION_STR = os.environ['SERVICE_BUS_CONNECTION_STR']
+QUEUE_NAME = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+
+async def main():
+    servicebus_client = ServiceBusClient.from_connection_string(conn_str=CONNECTION_STR)
+
+    async with servicebus_client:
```

Comment:
I've noticed that we're not super consistent with how much we set up before we sample a receive.

E.g. for "Receive_queue" sample we don't send before trying to receive.
But for "receive_deferred" we receive, defer, then receive deffered.  

I can kinda hand-wave that these are consistent, since in neither case are we sending, but figured I'd mull over this out loud in case you had thoughts on if these are good without setup.  Feel free to wont-fix, this isn't really an actionable comment more as it is me using this as an opportunity to ponder if our samples are representative enough.

Reply:
as per discussion OOB, we could open separate issues to focus on sample improvement